GNUmakefile

```diff
@@ -5,14 +5,16 @@ COUNT?=1
 TEST?=$(shell go list ./...)
 HASHICORP_PACKER_PLUGIN_SDK_VERSION?=$(shell go list -m github.com/hashicorp/packer-plugin-sdk | cut -d " " -f2)
 
-.PHONY: dev
-
+.PHONY: build
 build:
-	@go build -o ${BINARY}
+	@go generate main.go
+	@go build -o $(BINARY)
+
+.PHONY: install
+install: build
+	@packer plugins install --path ${BINARY} "github.com/cirruslabs/$(NAME)"
 
-dev:
-	go build -ldflags="-X '${BINARY}/version.VersionPrerelease=dev'" -o ${BINARY}
```

Comment:
Can you elaborate? 

You prefer to not use a generator (`go-gitver`) to resolve the version from the git tag? 

Do you propose to have logic in the Makefile to resolve the git tag, do some shell magic to resolve the next patch version, but only if the tree is dirty, and pass that as linker flags?

Reply:
> You prefer to not use a generator (`go-gitver`) to resolve the version from the git tag?

Yes, as I've mentioned in the comment above, at this point I don't think we need to `go:generate` anything because the Git tags serve us well.

We could probably get away with something as simple as:

```patch
diff --git a/GNUmakefile b/GNUmakefile
index 60c9c3b..7e00a60 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -4,6 +4,7 @@ BINARY=packer-plugin-${NAME}
 COUNT?=1
 TEST?=$(shell go list ./...)
 HASHICORP_PACKER_PLUGIN_SDK_VERSION?=$(shell go list -m github.com/hashicorp/packer-plugin-sdk | cut -d " " -f2)
+VERSION=$(shell git describe --tags --abbrev=0 | sed 's/^v//' | awk -F '.' '{printf "%d.%d.%d", $$1, $$2, $$3+1}')
 
 .PHONY: dev
 
@@ -11,7 +12,7 @@ build:
        @go build -o ${BINARY}
 
 dev:
-       go build -ldflags="-X '${BINARY}/version.VersionPrerelease=dev'" -o ${BINARY}
+       go build -ldflags="-X '${BINARY}/version.Version=${VERSION}' -X '${BINARY}/version.VersionPrerelease=dev'" -o ${BINARY}
        packer plugins install --path ${BINARY} "github.com/cirruslabs/$(NAME)"
 
 test:
```

Or call a script to properly handle the edge-cases. Another option is to invoke the `go run`, but that might be an overkill.

Reply:
I'm not sure I agree that re-implementing `go-gitver` in bash, including the various corner cases, is better than a simple generator. But if the generator is the problem we can combine the two:

`VERSION=$(go run git.rootprojects.org/root/go-gitver/v2@v2.0.2 version | awk 'NR==2')`

We shouldn't set `VersionPrerelease` in this case.

Reply:
Sigh, never mind, `version` prints the version of `go-gitver`.

Reply:
The bash version fails to account for `v1.11.0`, which should not have `dev`. The command should probably be `git describe --tags --dirty --always`, replacing anything after `-` with `-dev`, and only then bumping the patch version, otherwise leaving it alone. My `sed`/`awk` foo is not great enough to formulate that 😉

Reply:
I'm sure we can iterate on the bash script further. But as a wise person once said 😅

> Just want to make sure that we're not solving a non-existent problem here.

Is replacing `go-gitver` with bash magic solving a problem that using a generator has?

Reply:
> Is rewriting `go-gitver` in bash magic solving a problem that using a generator has?

Sure, it reduces the overall complexity: the changeset becomes smaller, it is more composable (you only need to change one line to play with versioning) and no new code is ever generated, so it's easier to reason about and build with `go build`.

There's also a value in keeping the project structure similar to [hashicorp/packer-plugin-scaffolding](https://github.com/hashicorp/packer-plugin-scaffolding) (and thus many other plugins), which makes it easier for the new contributors to figure things out.

I agree that there are great use-cases for `go:generate`, for example, to generate a bunch of ORM-like methods from an SQL file, which otherwise would be too error-prone if done by hand.

But the issue you're trying to solve here, namely, injecting a version string into the code at build time, does not require the use of `go:generate` at all and could be accomplished by much simpler means.

Reply:
Okey, I will push an updated PR shortly.